app/components/Settings/Settings.react.js

```diff
@@ -185,7 +185,10 @@ class Settings extends Component {
                 getTables();
             }
             if (tablesRequest.status === 200 && !selectedTable) {
-                setTable(head(tablesRequest.content));
+                // Don't set the table to be undefined; causes inf loop.
+                if (!isEmpty(tablesRequest.content)) {
+                    setTable(head(tablesRequest.content));
+                }
```

Comment:
this might be clearer as just:
```
if (tablesRequest.status === 200 && !isEmpty(tablesRequest.content) && !selectedTable) {
     setTable(head(tablesRequest.content));
}
```

in particular the order of the conditionals in the `if` statement is sort of the order about how I think of these things and their dependencies